csharp/HOLMS.Platform/HOLMS.Platform/Application.Client/ApplicationClient.cs

```diff
@@ -141,6 +141,19 @@ public class ApplicationClient : IDisposable, IApplicationClient {
             _clientId = oauth2ClientId;
         }
 
+        /// <summary>
+        /// Used to update the server connection profile. Disposes existing session.
+        /// </summary>
+        /// <param name="config"></param>
+        /// <param name="oauth2ClientId"></param>
+        public virtual void UpdateConfiguration(IApplicationClientConfig config, string oauth2ClientId) {
```

Comment:
I would approach this a bit differently. Why not just invalidate the old _ac, then create a new one?

Invalidate to ensure any remaining references to it aren't usable, and create a new one? Seems simpler to me, let me know what you think.